From f9c9f5b62107ae12bc052e4f288921f03efdb413 Mon Sep 17 00:00:00 2001
From: Bryan Brattlof <bb@ti.com>
Date: Tue, 1 Feb 2022 14:37:29 -0600
Subject: [PATCH 3/8] soc: j721s2: add host description information

This provides information of Host IDs that are permitted in the j721s2
SoC. These host IDs represent processing entities (or PEs) which is
mandatory identification of a Host in a processor.

Signed-off-by: Bryan Brattlof <bb@ti.com>
---
 Makefile                      |  3 +-
 common/socinfo.c              |  3 ++
 soc/j721s2/j721s2_host_info.c | 66 +++++++++++++++++++++++++++++++++
 soc/j721s2/j721s2_host_info.h | 70 +++++++++++++++++++++++++++++++++++
 4 files changed, 141 insertions(+), 1 deletion(-)
 create mode 100644 soc/j721s2/j721s2_host_info.c
 create mode 100644 soc/j721s2/j721s2_host_info.h

diff --git a/Makefile b/Makefile
index 74654cc..a33dfc6 100644
--- a/Makefile
+++ b/Makefile
@@ -118,7 +118,8 @@ AM62XSOURCES =\
 	      soc/am62x/am62x_rm_info.c \
 	      soc/am62x/am62x_sec_proxy_info.c
 
-J721S2SOURCES = soc/j721s2/j721s2_devices_info.c
+J721S2SOURCES = soc/j721s2/j721s2_devices_info.c \
+	       soc/j721s2/j721s2_host_info.c
 
 COMMONOBJECTS=	$(COMMONSOURCES:.c=.o)
 AM65XOBJECTS=	$(AM65XSOURCES:.c=.o)
diff --git a/common/socinfo.c b/common/socinfo.c
index 5fb5c17..fc291d2 100644
--- a/common/socinfo.c
+++ b/common/socinfo.c
@@ -74,6 +74,7 @@
 #include <soc/am62x/am62x_rm_info.h>
 #include <soc/am62x/am62x_sec_proxy_info.h>
 #include <soc/j721s2/j721s2_devices_info.h>
+#include <soc/j721s2/j721s2_host_info.h>
 
 /* Assuming these addresses and definitions stay common across K3 devices */
 #define CTRLMMR_WKUP_JTAG_DEVICE_ID	0x43000018
@@ -155,6 +156,8 @@ static void j721s2_init(void)
 {
 	struct ti_sci_info *sci_info = &soc_info.sci_info;
 
+	sci_info->host_info = j721s2_host_info;
+	sci_info->num_hosts = J721S2_MAX_HOST_IDS;
 	sci_info->devices_info = j721s2_devices_info;
 	sci_info->num_devices = J721S2_MAX_DEVICES;
 	soc_info.host_id = DEFAULT_HOST_ID;
diff --git a/soc/j721s2/j721s2_host_info.c b/soc/j721s2/j721s2_host_info.c
new file mode 100644
index 0000000..1fd9c2d
--- /dev/null
+++ b/soc/j721s2/j721s2_host_info.c
@@ -0,0 +1,66 @@
+/*
+ * J721S2 Hosts Info
+ *
+ * Copyright (C) 2022 Texas Instruments Incorporated - https://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include <tisci.h>
+#include <socinfo.h>
+
+struct ti_sci_host_info j721s2_host_info[] = {
+	[0]	= {0,	"TIFS",		"Secure",	"Security Controller"},
+	[1]	= {254,	"DM",		"Non Secure",	"Device Management"},
+	[2]	= {3,	"MCU_0_R5_0",	"Non Secure",	"Cortex R5 context 0 on MCU island"},
+	[3]	= {4,	"MCU_0_R5_1",	"Secure",	"Cortex R5 context 1 on MCU island(Boot)"},
+	[4]	= {5,	"MCU_0_R5_2",	"Non Secure",	"Cortex R5 context 2 on MCU island"},
+	[5]	= {6,	"MCU_0_R5_3",	"Secure",	"Cortex R5 context 3 on MCU island"},
+	[6]	= {10,	"A72_0",	"Secure",	"Cortex A72 context 0 on Main island"},
+	[7]	= {11,	"A72_1",	"Secure",	"Cortex A72 context 1 on Main island"},
+	[8]	= {12,	"A72_2",	"Non Secure",	"Cortex A72 context 2 on Main island"},
+	[9]	= {13,	"A72_3",	"Non Secure",	"Cortex A72 context 3 on Main island"},
+	[10]	= {14,	"A72_4",	"Non Secure",	"Cortex A72 context 4 on Main island"},
+	[11]	= {20,	"C7X_0_0",	"Secure",	"C7x_0 Context 0 on Main island"},
+	[12]	= {21,	"C7X_0_1",	"Non Secure",	"C7x_0 context 1 on Main island"},
+	[13]	= {22,	"C7X_1_0",	"Secure",	"C7x_1 Context 0 on Main island"},
+	[14]	= {23,	"C7X_1_1",	"Non Secure",	"C7x_1 context 1 on Main island"},
+	[15]	= {30,	"GPU_0",	"Non Secure",	"BXS context 0 on Main island"},
+	[16]	= {35,	"MAIN_0_R5_0",	"Non Secure",	"Cortex R5_0 context 0 on Main island"},
+	[17]	= {36,	"MAIN_0_R5_1",	"Secure",	"Cortex R5_0 context 1 on Main island"},
+	[18]	= {37,	"MAIN_0_R5_2",	"Non Secure",	"Cortex R5_0 context 2 on Main island"},
+	[19]	= {38,	"MAIN_0_R5_3",	"Secure",	"Cortex R5_0 context 3 on MCU island"},
+	[20]	= {40,	"MAIN_1_R5_0",	"Non Secure",	"Cortex R5_1 context 0 on Main island"},
+	[21]	= {41,	"MAIN_1_R5_1",	"Secure",	"Cortex R5_1 context 1 on Main island"},
+	[22]	= {42,	"MAIN_1_R5_2",	"Non Secure",	"Cortex R5_1 context 2 on Main island"},
+	[23]	= {43,	"MAIN_1_R5_3",	"Secure",	"Cortex R5_1 context 3 on MCU island"},
+	[24]	= {250,	"DM2TIFS",	"Secure",	"DM to TIFS communication"},
+	[25]	= {251,	"TIFS2DM",	"Non Secure",	"TIFS to DM communication"},
+	[26]	= {253,	"HSM",		"Secure",	"HSM Controller"},
+};
diff --git a/soc/j721s2/j721s2_host_info.h b/soc/j721s2/j721s2_host_info.h
new file mode 100644
index 0000000..180dd59
--- /dev/null
+++ b/soc/j721s2/j721s2_host_info.h
@@ -0,0 +1,70 @@
+/*
+ * J721S2 Host Info
+ *
+ * Copyright (C) 2022 Texas Instruments Incorporated - https://www.ti.com/
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *    Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ *
+ *    Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the
+ *    distribution.
+ *
+ *    Neither the name of Texas Instruments Incorporated nor the names of
+ *    its contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ *  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ *  OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ *  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ *  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ *  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ *  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ *  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ *  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef __J721S2_HOST_INFO_H
+#define __J721S2_HOST_INFO_H
+
+#define J721S2_HOST_ID_TIFS			0
+#define J721S2_HOST_ID_DM			254
+#define J721S2_HOST_ID_MCU_0_R5_0		3
+#define J721S2_HOST_ID_MCU_0_R5_1		4
+#define J721S2_HOST_ID_MCU_0_R5_2		5
+#define J721S2_HOST_ID_MCU_0_R5_3		6
+#define J721S2_HOST_ID_A72_0			10
+#define J721S2_HOST_ID_A72_1			11
+#define J721S2_HOST_ID_A72_2			12
+#define J721S2_HOST_ID_A72_3			13
+#define J721S2_HOST_ID_A72_4			14
+#define J721S2_HOST_ID_C7X_0_0			20
+#define J721S2_HOST_ID_C7X_0_1			21
+#define J721S2_HOST_ID_C7X_1_0			22
+#define J721S2_HOST_ID_C7X_1_1			23
+#define J721S2_HOST_ID_GPU_0			30
+#define J721S2_HOST_ID_MAIN_0_R5_0		35
+#define J721S2_HOST_ID_MAIN_0_R5_1		36
+#define J721S2_HOST_ID_MAIN_0_R5_2		37
+#define J721S2_HOST_ID_MAIN_0_R5_3		38
+#define J721S2_HOST_ID_MAIN_1_R5_0		40
+#define J721S2_HOST_ID_MAIN_1_R5_1		41
+#define J721S2_HOST_ID_MAIN_1_R5_2		42
+#define J721S2_HOST_ID_MAIN_1_R5_3		43
+#define J721S2_HOST_ID_DM2TIFS			250
+#define J721S2_HOST_ID_TIFS2DM			251
+#define J721S2_HOST_ID_HSM			253
+
+#define J721S2_MAX_HOST_IDS			27
+
+extern struct ti_sci_host_info	j721s2_host_info[];
+
+#endif /* __J721S2_HOST_INFO_H */
-- 
2.17.1

